vanilla-todo/src/core/Component.js

```diff
@@ -10,39 +10,54 @@ export default class Component {
     this.#state = observable(this.initState());
     observe(() => {
       this.render();
-      this.declare();
       this.setEvent();
+      this.mounted();
     });
   }
-  initState() {
-    return {};
-  }
   get state() {
-    return this.#state;
+    return Object.freeze({ ...this.#state });
   }
   get props() {
-    return this.#props;
+    return Object.freeze({ ...this.#props });
   }
   setState(newState) {
     for (const [key, value] of Object.entries(newState)) {
-      if (this.#state[key] === undefined) continue;
+      if (!this.#state.hasOwnProperty(key)) continue;
       this.#state[key] = value;
     }
   }
-  html() {
-    return `<div></div>`;
-  }
   render() {
-    this.$root.innerHTML = this.html();
+    if (this.html()) {
+      this.$root.innerHTML = this.html();
+    }
```

Comment:
렌더링이 중복으로 된다고 말씀하신게,
아마 이 부분 때문에 그런 것 같아요!
`html()`이라는 메소드를 두 번 실행 중이네요 ㅎㅎ 
render는 한 번 실행되지만, html은 두 번 실행되고
console을 html() 내부에 찍어놔서 중복으로 된다고 생각하신게 아닐까요?

```suggestion
    this.$root.innerHTML = this.html() || '';
```

이렇게 해주면 해결되지 않을까 싶어요!